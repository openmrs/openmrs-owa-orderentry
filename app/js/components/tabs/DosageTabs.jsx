import React, { Component } from 'react';
import PropTypes from 'prop-types';
import DosageTab from '../tabs/DosageTab';

class DosageTabs extends Component {
  handleTabClick = (tabIndex, tabName) => {
    this.props.handleFormTabs(tabIndex);
    this.props.formType(tabName);
  }

  renderChildrenWithTabsAsProps =() => React.Children.map(this.props.children, (child, index) => React.cloneElement(child, {
    onClick: this.handleTabClick,
    tabIndex: index,
    isActive: index === this.props.activeTabIndex,
  }))

  renderActiveTabContent=() => {
    const { children, activeTabIndex } = this.props;
    if (children[activeTabIndex]) {
      return children[activeTabIndex].props.children;
    }
  }

  render() {
    return (
      <div className="DosageTabs">
        Instructions:&nbsp;
        {this.renderChildrenWithTabsAsProps()}
        {this.renderActiveTabContent()}
      </div>
    );
  }
}

DosageTabs.propTypes = {
  children: PropTypes.array.isRequired,
  handleFormTabs: PropTypes.func.isRequired,
  formType: PropTypes.func.isRequired,
  activeTabIndex: PropTypes.number.isRequired,
};

export default DosageTabs;
