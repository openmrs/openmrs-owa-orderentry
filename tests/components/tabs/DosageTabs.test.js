import React from 'react';
import DosageTabs from '../../../app/js/components/tabs/DosageTabs'

describe('Dosage Tab component', () => {
    it('should render without crashing', () => {
        const wrapper = shallow(<DosageTabs/>);
        expect(wrapper).toMatchSnapshot();
    });
    it('should display the tab links', ()=>{
      const wrapper = shallow(<DosageTabs/>);
      wrapper.setState({ dosingDetails: { tabName: 'Standard', dosingType: 'org.openmrs.SimpleDosingInstructions' }});
      expect(wrapper.find('a')).toHaveLength(2);
    })     
});
