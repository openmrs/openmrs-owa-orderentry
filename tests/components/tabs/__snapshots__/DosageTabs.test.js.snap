// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dosage Tab component should render without crashing 1`] = `
<div
  className="DosageTabs"
>
  Instructions:Â 
</div>
`;
